Add chrome extension docs (#1004)
clerk/clerk-docs: docs/references/chrome-extension/sync-host.mdx

This guide assumes that you have followed the [Chrome Extension Quickstart](/docs/quickstarts/chrome-extension) and then the [Add React Router](/docs/references/chrome-extension/add-react-router) guide.

<Steps>
  ### Add `PLASMO_PUBLIC_SYNC_HOST` to your environment variables

  The `PLASMO_PUBLIC_SYNC_HOST` environment variable defines the host that the Chrome Extension will sync with.

  The values for `PLASMO_PUBLIC_SYNC_HOST` will differ between development and production environments. Using separate `.env.development` and `.env.production` files allows you to seamlessly pass the appropriate values to your builds.

  Use the following tabs to view the instructions for development versus production instances.

  <Tabs items={["Development", "Production"]}>
    <Tab>
      Add `PLASMO_PUBLIC_SYNC_HOST` to your `.env.development` file. The value should be `http://localhost`.

      ```env {{ filename: '.env.development', mark: [3] }}
      PLASMO_PUBLIC_CLERK_PUBLISHABLE_KEY={{pub_key}}
      CLERK_FRONTEND_API=https://{{fapi_url}}

>       PLASMO_PUBLIC_SYNC_HOST=http://localhost

Reviewer: ```suggestion
      PLASMO_PUBLIC_CLERK_SYNC_HOST=http://localhost
```

      ```
    </Tab>

    <Tab>
      Add `PLASMO_PUBLIC_SYNC_HOST` to your `.env.production` file. The value should be the domain your web app's production server runs on. For example, `https://clerk.com`.

      ```env {{ filename: '.env.production', mark: [3] }}
      PLASMO_PUBLIC_CLERK_PUBLISHABLE_KEY=pk_live_123
      CLERK_FRONTEND_API=https://yourdomain.com
      PLASMO_PUBLIC_SYNC_HOST=https://yourdomain.com
      ```
    </Tab>
  </Tabs>

  ### Add `syncHost` prop to your `<ClerkProvider>`

  Add the `syncHost` prop to your Chrome Extension's `<ClerkProvider>` component. This prop tells the `<ClerkProvider>` which host to sync with.

  ```tsx {{ filename: 'src/popup/layouts/root-layout.tsx', mark: [5, [7, 11], 22] }}
  import { ClerkProvider, SignedIn, SignedOut, UserButton } from '@clerk/chrome-extension'
  import { Link, Outlet, useNavigate } from 'react-router-dom'

  const PUBLISHABLE_KEY = process.env.PLASMO_PUBLIC_CLERK_PUBLISHABLE_KEY
  const SYNC_HOST = process.env.PLASMO_PUBLIC_SYNC_HOST

  if (!PUBLISHABLE_KEY || !SYNC_HOST) {
    throw new Error(

>       'Please add the PLASMO_PUBLIC_CLERK_PUBLISHABLE_KEY and PLASMO_PUBLIC_SYNC_HOST to the .env.development file',

Reviewer: ```suggestion
      'Please add the PLASMO_PUBLIC_CLERK_PUBLISHABLE_KEY and PLASMO_PUBLIC_CLERK_SYNC_HOST to the .env.development file',
```

    )
  }

  export const RootLayout = () => {
    const navigate = useNavigate()

    return (
      <ClerkProvider
        routerPush={(to) => navigate(to)}
        routerReplace={(to) => navigate(to, { replace: true })}
        publishableKey={PUBLISHABLE_KEY}
        afterSignOutUrl="/"
        syncHost={SYNC_HOST}
      >
        <div className="plasmo-w-[785px] plasmo-h-[600px]">
          <main>
            <Outlet />
          </main>
          <footer>
            <SignedIn>
              <Link to="/settings">Settings</Link>
              <UserButton />
            </SignedIn>
            <SignedOut>
              <Link to="/">Home</Link>
              <Link to="/sign-in">Sign In</Link>
              <Link to="/sign-up">Sign Up</Link>
            </SignedOut>
          </footer>
        </div>
      </ClerkProvider>
    )
  }
  ```

  ### Configure `host_permissions`

  `host_permissions` specifies which hosts, or websites, will have permission to sync auth state with your app. It accepts an array, allowing you to add more than one host.

>   In the `package.json` file, in the `manifest` object, update the `host_permissions` array. Remove `http://localhost/*` and replace with `$PLASMO_PUBLIC_SYNC_HOST/*`, as shown in the following example:

Reviewer: ```suggestion
  In the `package.json` file, in the `manifest` object, update the `host_permissions` array. Remove `http://localhost/*` and replace with `$PLASMO_PUBLIC_CLERK_SYNC_HOST/*`, as shown in the following example:
```

  ```json {{ filename: 'package.json', mark: [6] }}
  {
    // The rest of your package.json file
    "manifest": {
      "key": "$CRX_PUBLIC_KEY",
      "permissions": ["cookies", "storage"],

>       "host_permissions": ["$PLASMO_PUBLIC_SYNC_HOST/*", "$CLERK_FRONTEND_API/*"]

Reviewer: ```suggestion
      "host_permissions": ["$PLASMO_PUBLIC_CLERK_SYNC_HOST/*", "$CLERK_FRONTEND_API/*"]
```

---

> title: Sync your Chrome Extension with your web app

Author: Same here as in https://github.com/clerk/clerk-docs/pull/1004/commits/ff92b58598243b8e08fb9704303a766619840ded#r1809419432

With this page I'm just thinking of making the title clearer for people using search or skimming the side bar.

Reviewer 2: can do!

Reviewer 2: updated! [b089102](https://github.com/clerk/clerk-docs/pull/1004/commits/b089102b7f14df60439d79fd7c6313e1a735050a)

This guide assumes that you have followed the [Chrome Extension Quickstart](/docs/quickstarts/chrome-extension) and then the [Add React Router](/docs/references/chrome-extension/add-react-router) guide.

<Steps>
  ### Add `PLASMO_PUBLIC_SYNC_HOST` to your environment variables

  The `PLASMO_PUBLIC_SYNC_HOST` environment variable defines the host that the Chrome Extension will sync with.

  The values for `PLASMO_PUBLIC_SYNC_HOST` will differ between development and production environments. Using separate `.env.development` and `.env.production` files allows you to seamlessly pass the appropriate values to your builds.

  Use the following tabs to view the instructions for development versus production instances.

  <Tabs items={["Development", "Production"]}>
    <Tab>
      Add `PLASMO_PUBLIC_SYNC_HOST` to your `.env.development` file. The value should be `http://localhost`.

      ```env {{ filename: '.env.development', mark: [3] }}
      PLASMO_PUBLIC_CLERK_PUBLISHABLE_KEY={{pub_key}}

>       CLERK_FRONTEND_API={{fapi_url}}

Author: CLERK_FRONTEND_API=http://{{fapi_url}}

Reviewer 2: ```suggestion
      CLERK_FRONTEND_API=https://{{fapi_url}}
```

>       CLERK_FRONTEND_API={{fapi_url}}

Reviewer 2: ```suggestion
      CLERK_FRONTEND_API=https://{{fapi_url}}
```

      PLASMO_PUBLIC_SYNC_HOST=http://localhost
      ```
    </Tab>

    <Tab>
      Add `PLASMO_PUBLIC_SYNC_HOST` to your `.env.production` file. The value should be the domain your web app's production server runs on. For example, `https://clerk.com`.

      ```env {{ filename: '.env.production', mark: [3] }}

>       PLASMO_PUBLIC_CLERK_PUBLISHABLE_KEY={{pub_key}}

Reviewer 2: ```suggestion
      PLASMO_PUBLIC_CLERK_PUBLISHABLE_KEY=pk_live_123
```

>       CLERK_FRONTEND_API={{fapi_url}}

Reviewer 2: ```suggestion
      CLERK_FRONTEND_API=https://yourdomain.com
```

>       CLERK_FRONTEND_API={{fapi_url}}

Reviewer 2: good call - it seems we don't support production keys in the docs right now.

          <main>
            <Outlet />
          </main>
          <footer>
            <SignedIn>
              <Link to="/settings">Settings</Link>
              <UserButton />
            </SignedIn>
            <SignedOut>
              <Link to="/">Home</Link>
              <Link to="/sign-in">Sign In</Link>
              <Link to="/sign-up">Sign Up</Link>
            </SignedOut>
          </footer>
        </div>
      </ClerkProvider>
    )
  }
  ```

  ### Configure `host_permissions`

  `host_permissions` specifies which hosts, or websites, will have permission to sync auth state with your app. It accepts an array, allowing you to add more than one host.

  In the `package.json` file, in the `manifest` object, update the `host_permissions` array. Remove `http://localhost/*` and replace with `$PLASMO_PUBLIC_SYNC_HOST/*`, as shown in the following example:

  ```json {{ filename: 'package.json', mark: [6] }}
  {
    // The rest of your package.json file
    "manifest": {
      "key": "$CRX_PUBLIC_KEY",
      "permissions": ["cookies", "storage"],
      "host_permissions": ["$PLASMO_PUBLIC_SYNC_HOST/*", "$CLERK_FRONTEND_API/*"]
    }
  }
  ```

>   ### Add the Extension's ID to your web app's `allowed_origins`

Author: When setting the `.env` values above there were development and production sections but here there is only development. We should drop production above or add production here. That said, adding it here would have the same issue with the secret key here as the Publishable key above.

Reviewer 2: I've edited the instructions below, let me know what you think.
This will have to do until we support production keys in the docs.

Author: Looks good!

  To allow your Chrome Extension to sync with your web app, you must add the extension's ID to your web app's `allowed_origins`.

  > [!NOTE]
  > If you have not [configured a consistent key](/docs/references/chrome-extension/configure-consistent-key), you will have to repeat this step every time your extension's ID changes.

>   <SignedIn>
>     In your terminal, paste the following command and replace the `<CHROME_EXTENSION_KEY>` with your extension's ID.
>   </SignedIn>
> 

Reviewer 2: ```suggestion
```

>   <SignedOut>
>     1. In the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=api-keys), navigate to the **API Keys** page.
>     1. Copy your secret key.
>     1. In your terminal, paste the following command. Replace `YOUR_SECRET_KEY` with your Clerk secret key and the `<CHROME_EXTENSION_KEY>` with your extension's ID.
> 
>     The final result should resemble the following:
>   </SignedOut>

Reviewer 2: ```suggestion
    1. In the Clerk Dashboard, navigate to the [**API Keys**](https://dashboard.clerk.com/last-active?path=api-keys) page.
    1. Copy your secret key. It should start with `pk_test_` or `pk_live_` for your development and production instances, respectively.
    1. In your terminal, paste the following command. Replace `YOUR_SECRET_KEY` with your Clerk secret key and the `<CHROME_EXTENSION_KEY>` with your extension's ID.

    The final result should resemble the following:
```
